### ISSUE TITLE
what is correct way to pass class labels to confusion matrix

### ISSUE THREAD

**Issue by farazahmeds** [fepegar/torchio#211]
I'm trying to plot confusion matrix from predictions, final prediction tensor is concatenation of predictions appended during each validation run, `predlist = torch.cat([predlist, predicted.view(-1)])`.

but when I plot the class label axes go bad, i'm guessing since i'm appending labels `lbllist = torch.cat([lbllist, labels.view(-1)])` as well they show up on plot during each run.

![conf](https://user-images.githubusercontent.com/53582614/85900477-306a3900-b800-11ea-9cb3-e682d227abe7.png)

is there any way to pass class labels correctly in this case? 

my class labels are `class:0`  `class:1`  `class:2`

thanks.

ps: please excuse the figsize

_minimal reproducible example_

```

p1 = torchio.Subject(t1=torchio.Image('abc.nii.gz', mylabel=0))
p2 = torchio.Subject(t1=torchio.Image('xyz.nii.gz', mylabel=1))
p3 = torchio.Subject(t1=torchio.Image('cqy.nii.gz', mylabel=2))

subjects_list = [p1, p2, p3]

subjects_dataset = torchio.ImagesDataset(brats(), transform=transform)
dataloader = DataLoader(subjects_dataset, batch_size=1,shuffle=True, num_workers=num_workers)
```

```
##train step + confusion matrix


    with torch.no_grad():
        correct = 0
        total = 0
        for testdata in dataloader:
            images = testdata['t1']['data']).to('cuda:0')
            labels =   testdata['t1']['mylabel'].to('cuda:0')
            print(labels)

            outputs = model(images)

            _, predicted = torch.max(outputs.data, 1)
        # Append batch prediction results
        predlist = torch.cat([predlist, predicted.view(-1)])
        lbllist = torch.cat([lbllist, labels.view(-1)])

conf_mat=confusion_matrix(lbllist.cpu().numpy(), predlist.cpu().numpy())

plt.figure(figsize=(10,10))
plot_confusion_matrix(conf_mat, lbllist.cpu())
plt.show()
```

**Comment by fepegar** [comment]
I @farazahmeds, this is not related to TorchIO. I suggest you save `conf_mat ` and `lbllist.cpu()` and ask the [scikit-learn](https://scikit-learn.org/stable/support.html) folks using a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) using those variables.

Feel free to reopen if you consider the problem to be in TorchIO.
